numba_cuda/numba/cuda/core/sigutils.py

```diff
@@ -1,15 +1,124 @@
 # SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 # SPDX-License-Identifier: BSD-2-Clause
 
-from numba.core import types
-from numba.cuda import typing
+"""
+Utilities for signature normalization and type conversion.
+
+This module also provides type mapping between numba.core types
+and numba.cuda types to ensure cross-compatibility.
+"""
+
+from numba.cuda import types, typing
 
 try:
     from numba.core.typing import Signature as CoreSignature
+    from numba.core import types as core_types
```

Comment:
Not blocking for this PR but have we considered determining the presence of numba in a central way up front that we can just reference instead of proliferating try/catches? `if HAVE_NUMBA:` etc